tmt/cli.py

```diff
@@ -666,23 +666,29 @@ def export(
 #  Init
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+_init_template_choices = ['empty', 'mini', 'base', 'full']
+_init_templates = listed(_init_template_choices, join='or')
 @main.command()
 @click.pass_context
 @click.argument('path', default='.')
 @click.option(
-    '--mini', is_flag=True, help='Create a minimal plan.')
-@click.option(
-    '--base', is_flag=True, help='Create a plan and a test.')
-@click.option(
-    '--full', is_flag=True, help='Create a story, a plan and a test.')
+    '-t', '--template', default='empty', metavar='TEMPLATE',
+    type=click.Choice(_init_template_choices),
+    help='Template ({}).'.format(_init_templates), prompt='Template')
 @verbose_debug_quiet
```

Comment:
Prompt works nice. However there's one change in behaviour: You are now always asked for template:

    tmt init
    Template [empty]: 

In this case `empty` should be applied without asking. This will be quite common case I believe. Not sure if it's possible to instruct `click` to show prompt only if `-t` is provided.

Reply:
Makes sense. I believe click#549 means it's not possible without a deep dive into click internals, so I've removed the prompt altogether and made the output noisier in return in the name of discoverability:

```
$ tmt init
Tree '/home/asosedki/tmp/tmt' initialized.
To populate it with example content, use --template with mini, base or full.
```
```
$ tmt init -t full
Tree '/home/asosedki/tmp/tmt' already exists.
Applying template 'full'.
Directory '/home/asosedki/tmp/tmt/tests/example' created.
Test metadata '/home/asosedki/tmp/tmt/tests/example/main.fmf' created.
Test script '/home/asosedki/tmp/tmt/tests/example/test.sh' created.
Directory '/home/asosedki/tmp/tmt/plans' created.
Plan '/home/asosedki/tmp/tmt/plans/example.fmf' created.
Directory '/home/asosedki/tmp/tmt/stories' created.
Story '/home/asosedki/tmp/tmt/stories/example.fmf' created.
```

Reply:
Great! Looks now very beautiful ;-)